doc: add more details about squash merges (#1622)
danth/stylix: doc/src/commit_convention.md

The summary should start with a lowercase letter, and should not end with
punctuation.

> > [!TIP]
> > Adding `Co-authored-by` to your commits can be helpful in crediting people who
> > parshely authored the commit and will get special treement by github

Reviewer: What about this:

```suggestion
> [!TIP]
>
> Consider crediting co-authors with `Co-authored-by:` commit tags.
```

## Squash Merges

Most commits to `master` will also include a pull request number in brackets
after the summary. GitHub adds this automatically when creating a squash merge.

> Stylix maintainers may add relvent info to the description of these squash
> commits such as reviews, co-authors, relvent prs/issues and commits fixed by the
> current one, below is an example of this body:
> 
> ```
> stylix: guard home-manager-integration config (#1494)
> 
> Closes: https://github.com/nix-community/stylix/issues/1492
> Link: https://github.com/nix-community/stylix/pull/1494
> 
> Reviewed-by: Matt Sturgeon <matt@sturgeon.me.uk>
> Reviewed-by: awwpotato <awwpotato@voidq.com>
> Tested-by: https://github.com/alex-massa
> Reviewed-by: NAHO <90870942+trueNAHO@users.noreply.github.com>
> Tested-by: NAHO <90870942+trueNAHO@users.noreply.github.com>
> ```
> 
> > [!NOTE]
> > You do don't need to add these pieces of information to your commits, though
> > it is helpful to link relvent prs, issue and commits in your pr's description

Reviewer: What about just linking to a rather complete example:

```suggestion

See
[this](https://github.com/nix-community/stylix/commit/d73d8f6a4834716496bf8930a492b115cc3d7d17)
commit as an example.
```
